internal/manifests/targetallocator/deployment.go

```diff
@@ -22,16 +22,22 @@ import (
 
 	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
 	"github.com/open-telemetry/opentelemetry-operator/internal/config"
-	"github.com/open-telemetry/opentelemetry-operator/pkg/naming"
+	"github.com/open-telemetry/opentelemetry-operator/internal/reconcileutil/naming"
 )
 
 // Deployment builds the deployment for the given instance.
-func Deployment(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelemetryCollector) appsv1.Deployment {
+func Deployment(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelemetryCollector) *appsv1.Deployment {
 	name := naming.TargetAllocator(otelcol)
 	labels := Labels(otelcol, name)
-	annotations := Annotations(otelcol)
 
-	return appsv1.Deployment{
+	configMap, err := ConfigMap(cfg, logger, otelcol)
+	if err != nil {
+		logger.Info("failed to construct target allocator config map for annotations")
+		configMap = nil
```

Comment:
Should we return the Deployment here knowing the Pods will fail to be created (because the ConfigMap will be absent)? Philosophically, we should do one of the following imo:

1. Manifest generation is purely syntactic, and it's the reconciler's job to abort if one of them creates an error.
2. Manifest generation should return errors if it knows the output will fail semantically.

I think I'm ok with either, but it should be clear that the Deployment we return here may not actually work depending on other reconciliation results. In that case, returning it even though the ConfigMap fails is basically simplifying the API.

Reply:
I think we should error and set that on the status for the otelcol, preventing the bad thing from being rolled out and giving users the reason why. I think for now we should keep this the same as currently this change is a no-op (just shifting files around.) The next PR is going to change how we do mutation/reconciliation and is probably a better place for things like that IMO.